app/controllers/posts_controller.rb

```diff
@@ -2,7 +2,23 @@
 
 class PostsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_post, only: %i[destroy show update]
+  before_action :set_post, only: %i[destroy show update upvote downvote]
+
+  def index
+    @posts = Post.all.order(:desc)
+  end
+
+  def upvote
```

Comment:
[OPTIONAL]
Do you think that this is best approach?
Having a and upvote and downvote action imo is not the best approach. It's always best to have the basic actions.
Really this is created and destroy actions for the like controller.

Maybe I'm misunderstanding the use of this. 🤔 

Either way, maybe you can clean this up a bit.